### PR TITLE
Add the Conv1D support for NHWC format.

### LINKED ISSUE #81557: Bugs about permute function using MPS
### 🐛 Describe the bug

class FCN(nn.Module):
    def __init__(self):
        super(FCN, self).__init__()
        self.conv1 = nn.Conv1d(1, 128, 3)
        self.gap = nn.AdaptiveAvgPool1d(1)
        self.fc = nn.Linear(128, 2)

    def forward(self, x):
        batch_size = x.size(0)
        x = self.conv1(x)
        x = self.gap(x)
        x = x.view(batch_size, -1)
        x = self.fc(x)
        x = torch.log_softmax(x, dim=1)
        return x

if __name__ == '__main__':
    model = FCN().to('mps')
    out = model(torch.randn(128, 176, 1).permute(0, 2, 1).to('mps')).   # This line shows the error "Assertion failed: (mapIt != _jitValueTypes.end()), function getStaticType, file MPSRuntime_Project.h, line 435."
    
    # out = model(torch.randn(128, 1, 176).to('mps')).  # This could work well.

### Versions

PyTorch version: 1.13.0.dev20220714
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: version 3.22.1
Libc version: N/A

Python version: 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:25:14)  [Clang 12.0.1 ] (64-bit runtime)
Python platform: macOS-12.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.5
[pip3] torch==1.13.0.dev20220714
[pip3] torchaudio==0.14.0.dev20220603
[pip3] torchvision==0.14.0.dev20220714
[conda] torch                     1.13.0.dev20220714          pypi_0    pypi
[conda] torchaudio                0.14.0.dev20220603          pypi_0    pypi
[conda] torchvision               0.14.0.dev20220714          pypi_0    pypi

cc @kulinseth @albanD

### PR DESCRIPTION
**Pull request by kulinseth**
Fixes https://github.com/pytorch/pytorch/issues/81557

cc @DenisVieriu97

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/Convolution.mm [hunk 1]
```diff
@@ -61,14 +62,15 @@ Tensor _mps_convolution(
...
+  bool is_channels_last = (memory_format == at::MemoryFormat::ChannelsLast);
   auto output_t = at::empty(
                     conv_output_size(input->sizes(), weight->sizes(),
                                      padding, stride, dilation),
                     input->scalar_type(),
                     c10::nullopt,
                     kMPS,
                     c10::nullopt,
-                    memory_format);
+                    c10::nullopt);
```
Comment (albanD) [inline aten/src/ATen/native/mps/operations/Convolution.mm:73]: This means that the layout of the output will not be ChannelsLast anymore?

Comment (kulinseth) [reply aten/src/ATen/native/mps/operations/Convolution.mm:73]: right, so we add a Transpose at [line](https://github.com/pytorch/pytorch/pull/83121/files#diff-2b02a4fc9ee55c0e5e6c3ca50de5211843685b34b7f3fd709710b09504ccca78R166) to convert it to NCHW format.

**Review by malfet — APPROVED** [review]
